backend/product_management/context/controllers/responses/SneakerResponder.go

```diff
@@ -121,3 +122,50 @@ func SendSneakersByPagination(c *fiber.Ctx) error {
 		Page:     pageInt,
 	})
 }
+
+
+func SendRelatedProductsByTags(c *fiber.Ctx) error {
+	productID := c.Params("id")
+
+	var currentProduct models.Sneaker
+	objectID, err := primitive.ObjectIDFromHex(productID)
+	if err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid product ID"})
+	}
+
+	filter := bson.D{{Key: "_id", Value: objectID}}
+	err = database.SneakerCollection.FindOne(context.TODO(), filter).Decode(&currentProduct)
+	if err != nil {
+		return c.Status(fiber.StatusNotFound).JSON(fiber.Map{"error": "Product not found"})
+	}
+
+	var relatedProducts []models.Sneaker
+
+	filter = bson.D{
+		{"tags", bson.D{{"$in", currentProduct.Tags[0]}}},
+		{"tags", bson.D{{"$in", currentProduct.Tags[1]}}},
```

Comment:
Why not use the entire array instead of an item?

Reply:
Done